tensorflow_lite_support/python/task/text/text_searcher.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Text searcher task."""
+
+import dataclasses
+
+from tensorflow_lite_support.python.task.core.proto import base_options_pb2
+from tensorflow_lite_support.python.task.processor.proto import embedding_options_pb2
+from tensorflow_lite_support.python.task.processor.proto import search_options_pb2
+from tensorflow_lite_support.python.task.processor.proto import search_result_pb2
+from tensorflow_lite_support.python.task.text.pybinds import _pywrap_text_searcher
+
+_CppTextSearcher = _pywrap_text_searcher.TextSearcher
+_BaseOptions = base_options_pb2.BaseOptions
+_EmbeddingOptions = embedding_options_pb2.EmbeddingOptions
+_SearchOptions = search_options_pb2.SearchOptions
+
+
+@dataclasses.dataclass
+class TextSearcherOptions:
+  """Options for the text search task."""
+  base_options: _BaseOptions
+  embedding_options: _EmbeddingOptions = _EmbeddingOptions()
+  search_options: _SearchOptions = _SearchOptions()
+
+
+class TextSearcher(object):
+  """Class that performs embedding extraction on text, followed by
+  nearest-neighbor search in an index of embeddings through ScaNN."""
+
+  def __init__(self, options: TextSearcherOptions,
+               cpp_searcher: _CppTextSearcher) -> None:
+    """Initializes the `TextSearcher` object."""
+    # Creates the object of C++ TextSearcher class.
+    self._options = options
+    self._searcher = cpp_searcher
+
+  @classmethod
```

Comment:
@kinaryml Can you add a create_from_file method to align with other Python APIs? It creates the TextSearcher with the default options.

Reply:
To correct myself, I believe the index file is mandatory.